Custom Filter & JSON Logging Via Telemetry Subscribers

redixhumayun opened the pull request:
This PR:

* adds a builder method to allow supporting custom filter for the base layer
* switches out the json layer that was using the tracing bunyan formatter to use the json formatting layer provided by the tracing crate itself.

It can now be used like this

```
let (_guards, _filter_handle) = TelemetryConfig::new("my-app")
            .with_json_logs()
            .with_log_file("logs.json")
            .with_custom_filter(my_custom_filter)
            .with_layer(my_custom_layer)
            .init();
        std::mem::forget(_guards);
        std::mem::forget(_filter_handle);
```

velvia reviewed (approved):
@redixhumayun this looks good, can you please update the README as well, I think it has some reference to Bunyan, just need to change that to be consistent, then I'll merge it.  Thanks

redixhumayun commented:
> @redixhumayun this looks good, can you please update the README as well, I think it has some reference to Bunyan, just need to change that to be consistent, then I'll merge it. Thanks

@velvia I've updated the docs and also removed the crate from the `cargo.toml` file. Would probably be confusing to people why it was even required since it's no longer being used. 

Let me know if this is good.
